lib/superset/dashboard/filters/list.rb

```diff
@@ -0,0 +1,27 @@
+module Superset
+  module Dashboard
+    module Filters
+      class List < Superset::Request
+        attr_reader :id
+
+        def initialize(id)
+          @id = id
+        end
+
+        def perform
+          filters_configuration = JSON.parse(dashboard.result['json_metadata'])['native_filter_configuration'] || []
+          return Array.new unless filters_configuration && filters_configuration.any?
+  
+          # pull only the filters dataset ids from the dashboard
+          filters_configuration.map { |c| c['targets'] }.flatten.compact.map { |c| c['datasetId'] }.flatten.compact
+        end
```

Comment:
@soundarya-mv  I dont thing we  need a new class for this.

We can just add another method to the Dashboard::Get class

https://github.com/rdytech/superset-client/blob/develop/lib/superset/dashboard/get.rb#L25
```
      def json_metadata
        JSON.parse(result['json_metadata'])
      end

     def filter_configuration
         json_metadata['native_filter_configuration'] || []
     end

```